src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -9922,7 +9942,7 @@ void func_808473D4(PlayState* play, Player* this) {
     }
 }
 
-s32 func_80847A78(Player* this) {
+s32 Player_UpdateHoverBoots(Player* this) {
```

Comment:
This name does make sense given it's almost entirely processing hover-boots code and might be fine, but part of me is thinking this function is a more general `Player_TryLeavingGround` sort of function.

Just to add a perspective (although it doesn't need to direct naming this function), you have an identical function in MM used with the same logic in `Player_ProcessSceneCollision` That doesn't deal with hover boots, just the more general "try to leave the ground" approach, that returns true if successful and false if still on the ground

https://github.com/engineer124/mm/blob/c9796930fd528f4aeaff5a248c58a1d35859eebd/src/overlays/actors/ovl_player_actor/z_player.c#L11104-L11120

Reply:
Yeah I was aware of MM's refactors when naming this. Still think oots is entirely hoverboots oriented enough to warrant the name. The return values can be viewed as just concerning whether hes left the ground or not. But thats ignoring basically everything else the function does, so its a matter of what you want to prioritize naming the function based on.

Even in MM's case im not sure "TryLeavingGround" makes sense. It does unset the ground flag in certain cases (in oot on certain floor types while wearing hover boots, in mm when youre deku about to bounce on water). But in 99% of cases, the "leaving the ground" part happens naturally by `UpdateBgCheckInfo` earlier on. Its more of a response to links current grounded state, as opposed to trying to control what it is

Reply:
I'd argue that a significant amount of state change is completely unrelated to the hover boots mechanic, enough to where it's misleading to call the function "UpdateHoverBoots".

Reply:
after some time to think about this, im more open to renaming it.

I do however think, mzx, that saying "significant amount of state change is completely unrelated to the hover boots mechanic" it not true at all. 
The only thing truly not related to the hoverboots is the two lines at the bottom, resetting the floor type and angles, and the return value itself. The state change involved of unsetting the bgcheck ground flag ONLY occurs if the hover boots related stuff has passed. normal "leaving the ground" happens elsewhere.

Phrased another way, yes the return value does represent in general "link has left the ground". However, the only way that flag gets unset in this function is if the hoverboots activate. If link left the ground "naturally", the flag would have been unset earlier in the bgcheck update process.

I do not know how to make this function name involve "leaving the ground" without making it sounds like it handle more than it does.

Reply:
> I do however think, mzx, that saying "significant amount of state change is completely unrelated to the hover boots mechanic" it not true at all. The only thing truly not related to the hoverboots is the two lines at the bottom, resetting the floor type and angles, and the return value itself. The state change involved of unsetting the bgcheck ground flag ONLY occurs if the hover boots related stuff has passed. normal "leaving the ground" happens elsewhere.

I mean, if you list out all external states that can potentially change state after the function is called...

```
this->hoverBootsTimer
sFloorType
sFloorPitchShape
this->floorPitch
this->floorPitchAlt
this->actor.bgCheckFlags
return
```

...it's clear there is only 1 variable dedicated to hoverboots vs 4 that you say yourself are completely unrelated, and then lastly bgCheckFlags and return just chillin' there. That seems like a significant amount of state being changed unrelated to hover boots.

To me, it seems possible that you're giving too much value to the amount of logic needed to perform the state change, rather than considering the totality of the states being changed. Yes, the majority of the function instruction wise is spent on hoverboots stuff, but only because setting the state of `hoverBootsTimer` and `actor.bgCheckFlags` correctly is more complicated than everything else.

Consider what purpose the function has when `this->currentBoots == PLAYER_BOOTS_HOVER` is false. hoverBootsTimer is assigned to 19, and all of the rest of the logic essentially gets tossed out. What remains is essentially this code:

```c
    if (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND) {
        return false;
    } else {
        sFloorType = FLOOR_TYPE_0;
        this->floorPitch = this->floorPitchAlt = sFloorPitchShape = 0;
        return true;
    }
```

And if you look at the two places where `func_80847A78`/`Player_UpdateHoverBoots` is called, it is NOT dependent on the state of bgCheckFlags. Both branches are reachable.

Yea, the function probably exists to be able to hook in and implement hovering, but strictly speaking it still does other things unrelated to hovering

Reply:
My point regarding `bgCheckFlags` is that its state is only changed if the hoverboots activate. The ground flag will not get unset for any other reason in this function, other than the hoverboots activating. To say that the state of bgcheck flags here is unrelated to hoverboots is just not correct. 

Yes, I do agree that the two lines of state change regarding floor type and floor angles do not regard hover boots, and will execute when off the ground no matter the state of the hoverboots. But its pretty clearly, in my opinion, not the main point of the function.

Reply:
and my point is that `bgCheckFlags` doesn't need to change within the function for the function to have side effect of "detaching" the player from the ground in some form.

Reply:
But it doesnt. The function doesnt unset the flag at all unless the hoverboots activate under the special condition. How else is it controlling players "detaching" from the ground?

Reply:
yeah that isnt detaching at all. 0 floor type is normal and used for most floors. its just resetting it to normal in the case you happen to be standing on a special floor that allows hoverboots. the long condition above in the function is dealing with activating hoverboots in special conditions while still standing on the floor, for example like standing on quick sand.

the floor angles get zeroed so the values arent stale later. but this is by no means doing any heavy lifting of link vs floor logic. The actual "detaching" of the floor is handled elsewhere, and is why I still say that this function is largely hover boots related

Reply:
@AngheloAlf my issue with `Player_TryHoveringWhenLeavingGround` is that this function isnt really dealing with the "leaving the ground" part. This happens elsewhere in player, correctly updating his ground flag to be unset when appropriate.

The special case dealt with in this function is like I mentioned in my previous comment, niche cases where player's hoverboots can activate while he is still standing on the ground. Like standing on quick sand. That block doesnt concern all cases of player leaving the ground and hoverboots activating.

That interaction is purely controlled by the hoverBootsTimer which may or may not be reset to a value of 19 depending on the state of the ground flag. Which is why I consider it more of an update for the hoverboots based on mostly preexisting state changes (other than the niche case mentioned)

Reply:
I added a function comment to hopefully help clear up the confusion about the return and what is and isnt related to hover boots. I still think the name I have fits best but am still open to discussion about changing it.